integrationtests/gitjob/controller/controller_test.go

```diff
@@ -542,10 +542,15 @@ var _ = Describe("GitJob controller", func() {
 					job.Status.StartTime = &metav1.Time{Time: time.Now().Add(-1 * time.Hour)}
 					job.Status.Conditions = []batchv1.JobCondition{
 						{
-							Type:    "Failed",
+							// using Stalled because the Compute function uses Stalled
+							// for returning the condition message and it's simpler.
+							// For testing it in a different way we would need to setup a more complex
+							// scenario defining the job pods
+							// We are simulating job failures.
+							Type:    "Stalled",
 							Status:  "True",
 							Reason:  "BackoffLimitExceeded",
-							Message: "Job has reached the specified backoff limit",
+							Message: `{"fleetErrorMessage":"fleet error message","level":"fatal","msg":"Fleet cli failed","time":"2025-04-15T14:53:15+02:00"}`,
```

Comment:
Beware that `message` in status conditions is by definition meant to be human-readable:
```
❯ kubectl explain job.status.conditions.message
GROUP:      batch
KIND:       Job
VERSION:    v1

FIELD: message <string>


DESCRIPTION:
    Human readable message indicating details about last transition.
```

Reply:
I think for jobs it's okay and it allows much better errors in the UI? 

Additionally:
 
> JSON is a text-based, human-readable format for representing ...